src/test/java/org/javarosa/entities/EntitiesTest.java

```diff
@@ -127,6 +127,42 @@ public void fillingFormWithCreate_makesEntityAvailable() throws IOException, XFo
         assertThat(entities.get(0).action, equalTo(EntityAction.CREATE));
     }
 
+    @Test
+    public void fillingFormWithCreate_withoutAnId_doesNotMakeEntityAvailable() throws IOException, XFormParser.ParseException {
+        Scenario scenario = Scenario.init("Create entity form", XFormsElement.html(
+            asList(
+                new Pair<>("entities", "http://www.opendatakit.org/xforms/entities")
+            ),
+            head(
+                title("Create entity form"),
+                model(asList(new Pair<>("entities:entities-version", "2022.1.1")),
+                    mainInstance(
+                        t("data id=\"create-entity-form\"",
+                            t("id"),
+                            t("meta",
+                                t("entity dataset=\"people\" create=\"1\" id=\"\"",
+                                    t("label")
+                                )
+                            )
+                        )
+                    ),
+                    bind("/data/id").type("string"),
+                    bind("/data/meta/entity/@id").type("string").calculate("/data/id"),
+                    bind("/data/meta/entity/label").type("string").calculate("/data/name")
+                )
+            ),
+            body(
+                input("/data/id")
+            )
+        ));
+
+        scenario.getFormEntryController().addPostProcessor(new EntityFormFinalizationProcessor());
+        scenario.finalizeInstance();
+
+        List<Entity> entities = scenario.getFormEntryController().getModel().getExtras().get(Entities.class).getEntities();
+        assertThat(entities.size(), equalTo(0));
```

Comment:
@lognaturel I considered exposing a blank ID to a client, but I'm not actually sure that makes sense. Given that `update` and `id` could both be conditional in a form, I'm not sure this is an exceptional case. It seems intuitive to me that if I should be able to write a form that conditionally updates based on whether an entity was ever selected or not for instance (like in this example form).

We could get strict and say that if `update` is truthy, then `id` cannot be null though? The [spec](https://getodk.github.io/xforms-spec/entities) would need some small updates for that, and I think if that's the case we'd want to fail to finalize (as the form would "invalid").